From 28c7ae21453677870c92cc800bd821bcd0bc2074 Mon Sep 17 00:00:00 2001
From: Oleksandr Shaporov <o.shaporov@inango-systems.com>
Date: Wed, 7 Apr 2021 23:09:41 +0300
Subject: [PATCH] Commit Title: Capability to define list of bridges via
 specialized symbols

BugID: 23612
Bug Fixed(y/N[blankN]): N
CodeReviewId: 0
Description: Added ability to define list of bridges via * and ! symbols.
---
 datapath/brcompat_main.c | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/datapath/brcompat_main.c b/datapath/brcompat_main.c
index 30c99ee91..440e19816 100644
--- a/datapath/brcompat_main.c
+++ b/datapath/brcompat_main.c
@@ -803,10 +803,21 @@ static bool check_bridge_list(const char *name)
 	for (i = 0; i < size_list; i++) {
 		if (!br_list[i])
 			break;
-		if (strcmp(name, br_list[i]) == 0)
-			return true;
+		if (br_list[i][0] == '!') {
+			if (strcmp(name, &br_list[i][1]) == 0)
+				return false;
+		} else {
+			if (strcmp(name, br_list[i]) == 0)
+				return true;
+		}
 	}
-	return false;
+	if (!br_list[0])
+		return false;
+
+	if (br_list[0][0] == '*') 
+		return true;
+
+		return false;
 }
 
 static int __init brc_init(void)
